src/ansys/aedt/core/modeler/circuits/primitives_maxwell_circuit.py

```diff
@@ -284,3 +284,87 @@ def create_winding(self, name=None, location=None, angle=0, use_instance_id_netl
         )
         id.set_property("Name", name)
         return id
+
+    @pyaedt_function_handler(compname="name")
+    def create_isin(self, name=None, value=1, location=None, angle=0, use_instance_id_netlist=False):
+        """Create a sinusoidal current source.
+
+        Parameters
+        ----------
+        name : str, optional
+            Name of the current source. The default is ``None``.
+        value : float, optional
+            Value for the amplitude of current. The default is ``1``.
+        location : list of float, optional
+            Position on the X axis and Y axis.
+        angle : float, optional
+            Angle of rotation in degrees. The default is ``0``.
+        use_instance_id_netlist : bool, optional
+            Whether to use the instance ID in the net list or not. The default is ``False``.
+
+        Returns
+        -------
+        :class:`ansys.aedt.core.modeler.cad.object_3dcircuit.CircuitComponent`
+            Circuit Component Object.
+
+        References
+        ----------
+
+        >>> oEditor.CreateComponent
+        """
+        if location is None:
```

Comment:
@DaveTwyman please also handle the case when name=None because otherwise the method will file. Generate a random name if it is None.

Reply:
@gmalinve , Good point, whilst it creates the component as expected when 'name=None' it does generate several error messages in the Message Manager window. When 'name = Newname' there are no errors. 

I see all the other components (Resistor, Inductor etc) in the same library have the same behavior when None is passed for the name. Looking at it in a bit more depth, the circuit editor schematic defaults to allow elements to be created with no name (unlike say boundaries or geometry in the 3D modeler of AEDT). I think this behavior should be looked at in more detail either creating a new random name as you've suggested or assigning no name as is the default for new components being added to Maxwell schematic.

Reply:
The None case is now supported for both new components and the others in this library. Support for names of type String, Integer and Float is now covered as per existing Maxwell Circuit Editor usage